src/extractors/gitmojiextractor.py

```diff
@@ -0,0 +1,56 @@
+"""
+Copied and changed from nerdextractor.py in https://github.com/fdw/rofimoji
+"""
+
+
+import html
+from pathlib import Path
+import json
+
+import requests
+
+from characterfactory import Character
+
+
+class GitmojiExtractor(object):
+    def __init__(self: 'GitmojiExtractor'):
+        self.icons = []
+
+    def fetch_icons(self: 'GitmojiExtractor'):
+        print("Downloading list of gitmojis")
+
+        response = requests.get(
+            "https://raw.githubusercontent.com/carloscuesta/gitmoji/master/src/data/gitmojis.json",
+            timeout=60
+        )
+        """
+        {
+            'emoji': '🎨',
+            'entity': '&#x1f3a8;',
+            'code': ':art:',
+            'description': 'Improve structure / format of the code.',
+            'name': 'art',
+            'semver': None
+        }
+        """
+        for char in response.json()["gitmojis"]:
+            icon = ord(char["emoji"][0])
+            name = char["description"]
+            self.icons.append(Character(icon, name))
+
+    def write_to_file(self: 'GitmojiExtractor'):
+        if len(self.icons) == 0:
```

Comment:
This is just a shortcut, right?

i'm not sure if it's easier to understand with the `if` than without... what do you say?

Reply:
Now this isn't my code, I don't know why this check was added. Best to keep the check there, I think.

Reply:
You mean you copied this from `nerdextractor.py`, right?

In any case, we can leave this, but I don't think it adds anything. If `self.icons` is empty, the `for` loop doesn't execute and thus nothing is written anyway.

Reply:
Yeah, makes sense. I'll just remove it.